applications/nrf_desktop/doc/leds.rst

```diff
@@ -0,0 +1,62 @@
+
+LED effect (:c:type:`led_effect`) is described by pointer to array of LED steps (:cpp:member:`steps`), size of the array (:cpp:member:`step_count`) and flag indicating if sequence should start over after if finishes (:cpp:member:`loop_forever`).
+
+To achieve set effect, LED color is upated periodically (based on defined steps).
+During every update (substep), the next LED color is calculated using a linear approximation between the current LED color and the color described in next step (:cpp:member:`color`).
+Single step also defines number of updates (substeps) for color change between the step and previous one  (:cpp:member:`substep_count`) and period of time between color updates (:cpp:member:`substep_time`).
+After achieving color described in next step, index of next step is updated.
+Afrer last step sequence starts over only if :cpp:member:`loop_forever` flag is set for given effect.
+
+Module Events
+*************
+
++------------------+---------------+-------------+------------------+-------------+
+| Source Module    | Input Event   | This Module | Output Event     | Sink Module |
++================+=================+=============+==================+=============+
+| :ref:`led_state` | ``led_event`` | :ref:`leds` |                  |             |
++------------------+-------------+----------------+-----------------+-------------+
+
+Configuration
+*************
+
+The module uses Zephyr PWM driver for setting LED color (brightness).
+To use the module, set the following configuration options:
+- :option:`CONFIG_PWM`
+- :option:`CONFIG_PWM_0`
+
+In :option:`CONFIG_PWM_0`, 0 corresponds to the PWM port number. Make sure to configure all PWM ports used by application.
+
+Module is enabled using :option:`CONFIG_DESKTOP_LED_ENABLE`.
+
+Additional configuration options may be used to configure:
+* number of LEDs is use (:option:`CONFIG_DESKTOP_LED_COUNT`),
+* number of color channels per diode (:option:`CONFIG_DESKTOP_LED_COLOR_COUNT`),
+* maximum value of LED brightness (:option:`CONFIG_DESKTOP_LED_BRIGHTNESS_MAX`).
+
+Symbols from :ref:`zephyr:_device-tree` are used to define mapping of the PWM channels to pin numbers.
+By default the symbols are defined in board configuration, but you can redefine them in ``dts.overlay`` file.
+Configuration for the board is written in the ``leds_def.h`` file.
+Both mentioned files are located in the board specific directory in application configuration folder.
+
+Implementation details
+**********************
+
+LED color is achieved by setting proper pulse widths for PWM signals.
+To achieve the set LED effect, colors for the given LED are periodically updated using work (:c:type:`k_delayed_work`).
+One work automatically updates color of single LED.
+
+This module turns off all LEDs when application goes to the power down state.
+In such case, the PWM drivers are set to the suspended state to reduce the power consumption.
+While the application goes to error state, LEDs are used to indicate error.
```

Comment:
```suggestion
If the application goes to the error state, the LEDs are used to indicate error.
```
How? Clarify in the doc what the behavior of the LEDs in the error state is...

Reply:
This is be defined in `led_state_def.h` (per board). I wanted to keep description of effects set to LEDs in `led_state` docs.